hack-sound-server

```diff
@@ -0,0 +1,117 @@
+        bus = self.pipeline.get_bus()
+        bus.add_signal_watch()
+        bus.connect("message", self.__bus_message_cb)
+
+    def play(self):
+        self.pipeline.set_state(Gst.State.PLAYING)
+
+    @property
+    def sound_location(self):
+        return os.path.join(get_sounds_dir(), self.metadata["sound-file"])
+
+    def _build_pipeline(self):
+        spipeline = "filesrc location=\"%s\" ! decodebin ! autoaudiosink" % self.sound_location
+        return Gst.parse_launch(spipeline)
+
+    def __bus_message_cb(self, unused_bus, message):
+        if message.type == Gst.MessageType.EOS:
+            self.emit("eos")
+
+
+class HackSoundServer(Gio.Application):
+
+    _DBUS_NAME = "com.endlessm.HackSoundServer"
+    _DBUS_XML = """
+    <node>
+      <interface name='com.endlessm.HackSoundServer'>
+        <method name='PlaySound'>
+          <arg type='s' name='sound_event' direction='in'/>
+          <arg type='s' name='uuid' direction='out'/>
+        </method>
+      </interface>
+    </node>
+    """
+
+    def __init__(self):
+        super().__init__(application_id=self._DBUS_NAME,
+                         flags=Gio.ApplicationFlags.IS_SERVICE)
+        self._dbus_id = None
+        with open(get_metadata_path(), "r") as metadata_file:
+            self.metadata = json.load(metadata_file)
+        self.hold()
+        self.players = {}
+
+    def do_dbus_register(self, connection, path):
+        info = Gio.DBusNodeInfo.new_for_xml(self._DBUS_XML)
+        self._dbus_id = connection.register_object(path,
+                                                   info.interfaces[0],
+                                                   self.__method_called_cb)
+        return Gio.Application.do_dbus_register(self, connection, path)
+
+    def do_dbus_unregister(self, connection, path):
+        Gio.Application.do_dbus_unregister(self, connection, path)
+        if not self._dbus_id:
+            return
+        connection.unregister_object(self._dbus_id)
+        self._dbus_id = None
+
+    def play_sound(self, sound_event_id, sender, invocation):
+        uuid_ = str(uuid.uuid4())
+        player = HackSoundPlayer(self.metadata[sound_event_id], sender)
+        self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
+                                              sender)
+        self.players[uuid_].connect("eos", self.__player_eos_cb, uuid_)
+        self.players[uuid_].play()
+        invocation.return_value(GLib.Variant('(s)', (uuid_, )))
+
+    def __method_called_cb(self, connection, sender, path, iface,
+                           method, params, invocation):
+        if method == "PlaySound":
+            self.play_sound(params[0], sender, invocation)
+
+    def __player_eos_cb(self, player, uuid_):
+        del self.players[uuid_]
+
+
+if __name__ == "__main__":
+    Gst.init(None)
+    server = HackSoundServer()
+    GLib.unix_signal_add(GLib.PRIORITY_DEFAULT, signal.SIGINT, server.release)
```

Comment:
To be clear, I think `SIGINT` should still stop the application immediately, and `SIGTERM` should stop it politely...